src/core/simulation_results.jl

```diff
@@ -118,26 +116,45 @@ IS.get_parameters(result::StageResults) = result.parameter_values
 #IS.get_total_cost(result::StageResults) = result.total_cost
 #IS.get_optimizer_log(results::StageResults) = results.optimizer_log
 
+_get_containers(x::StageResults) = (x.variable_values, x.parameter_values, x.dual_values)
+_get_dicts(res::StageResults) = (y for x in _get_containers(res) for y in values(x))
+
+function _get_store_value(
+    res::StageResults,
+    field::Symbol,
+    names::Vector{Symbol},
+    timestamps,
+    ::Nothing,
+)
+    simulation_store_path = joinpath(get_execution_path(res), "data_store")
+    return h5_store_open(simulation_store_path, "r") do store
+        _get_store_value(res, field, names, timestamps, store)
+    end
+end
+
 function _get_store_value(
     res::StageResults,
     field::Symbol,
     names::Vector{Symbol},
     timestamps,
+    store::SimulationStore,
```

Comment:
@jd-lara You should note this change. I changed the functions in the readX call stacks to accept an optional store. This will allow us to avoid opening/closing the store repeatedly but still guarantee that the file handle gets closed deterministicly.

Reply:
noted